eos-update-flatpak-repos

```diff
@@ -1132,21 +1132,24 @@ def rename_exports(repo, mtree, old_id, new_id, path='export'):
                                      GLib.KeyFileError.KEY_NOT_FOUND):
                         raise
 
-                # update the places inside the desktop file where the ID is embedded
-                for key in [GLib.KEY_FILE_DESKTOP_KEY_ICON,
-                            'X-Endless-SplashBackground']:
-                    try:
-                        value = desktop_file.get_string(
-                            GLib.KEY_FILE_DESKTOP_GROUP, key
-                        )
-                    except GLib.GError as e:
-                        if not e.matches(GLib.KeyFile.error_quark(),
-                                         GLib.KeyFileError.KEY_NOT_FOUND):
-                            raise
-                        continue
-
-                    value = value.replace(old_id, new_id)
-                    desktop_file.set_string(GLib.KEY_FILE_DESKTOP_GROUP, key, value)
+                # update the icon name which matches the ID
+                try:
+                    value = desktop_file.get_string(
+                        GLib.KEY_FILE_DESKTOP_GROUP,
+                        GLib.KEY_FILE_DESKTOP_KEY_ICON
+                    )
+                except GLib.GError as e:
+                    if not e.matches(GLib.KeyFile.error_quark(),
+                                     GLib.KeyFileError.KEY_NOT_FOUND):
+                        raise
+                    continue
```

Comment:
This is wrong, it will skip the x-flatpak-renamedfrom block. Move the 'replace and set' stuff into an `else:` block.

Reply:
.oO( how did the test pass? )

Reply:
Added a (failing) test case that catches this.

Reply:
(and a fix to be squashed)

Reply:
Lucky one of us is awake! :)